File: test/client_code/tests/echo/echo.py
"""
echo/echo.py
"""


def main(use):

    echo = use("@@/echo/")
    result = echo(42)
    print("result:", result)

    echo = use("@/echo").echo
    result = echo(42)
    print("result:", result)

    echo = use("/echo").echo
    result = echo(42)
    print("result:", result)

    echo = use("assets/echo.js").echo
    result = echo(42)
    print("result:", result)

    echo = use("rpc/echo", dict(spinner=True))
    response = echo(42)
    result = response.get("result")
    print("result:", result)

    server = use("@/server").server
    response = server.echo(42)
    result = response.result
    print("result:", result)

File: test/client_code/tests/echo/echo.js
/*
echo/echo.js
*/

export default async () => {
 

  await (async () => {
    const echo = await use("@@/echo/");
    const result = echo(42);
    console.log("result:", result);
  })();

  await (async () => {
    const echo = (await use("@/echo")).echo;
    const result = echo(42);
    console.log("result:", result);
  })();

  await (async () => {
    const echo = (await use("/echo")).echo;
    const result = echo(42);
    console.log("result:", result);
  })();

  await (async () => {
    const echo = (await use("assets/echo.js")).echo;
    const result = echo(42);
    console.log("result:", result);
  })();

  await (async () => {
    const echo = await use("rpc/echo", { spinner: true });
    const response = await echo(42);
    const result = response.result;
    console.log("result:", result);
  })();

  await (async () => {
    const { server } = await use("@/server");
    const response = await server.echo(42);
    //const result = response.result;
    //console.log("result:", result);
  })();
};
